src/auth/ejabberd_auth_anonymous.erl

```diff
@@ -153,10 +153,10 @@ remove_connection(SID, LUser, LServer) ->
 
 
 %% @doc Register connection
--spec register_connection(Acc :: map(),
+-spec register_connection(Acc,
                           SID :: ejabberd_sm:sid(),
                           JID :: jid:jid(),
-                          Info :: list()) -> map().
+                          Info :: list()) -> Acc when Acc :: any().
```

Comment:
why is it now less restrictive?

Reply:
Simply because it was never a map. This hook was run with `ejabberd_hooks:run` function which passes just atom `ok` as the `Acc` at this moment.